src/chainparams.cpp

```diff
@@ -246,6 +254,89 @@ class CTestNetParams : public CChainParams {
     }
 };
 
+/**
+ * SigNet
+ */
+class SigNetParams : public CChainParams {
+public:
+    SigNetParams(const ArgsManager& args) {
+        std::vector<uint8_t> bin;
+        vSeeds.clear();
+        uint32_t genesis_nonce = 0;
+
+        if (!args.IsArgSet("-signet_blockscript")) {
+            throw std::runtime_error(strprintf("%s: -signet_blockscript is mandatory for signet networks", __func__));
+        }
+        if (args.GetArgs("-signet_blockscript").size() != 1) {
+            throw std::runtime_error(strprintf("%s: -signet_blockscript cannot be multiple values.", __func__));
+        }
+        bin = ParseHex(args.GetArgs("-signet_blockscript")[0]);
+        if (args.IsArgSet("-signet_genesisnonce")) {
+            genesis_nonce = args.GetArg("-signet_genesisnonce", 0);
+        }
+        if (args.IsArgSet("-signet_seednode")) {
+            vSeeds = gArgs.GetArgs("-signet_seednode");
+        }
+
+        LogPrintf("SigNet with block script %s\n", gArgs.GetArgs("-signet_blockscript")[0]);
+
+        strNetworkID = "signet";
+        consensus.signet_blocks = true;
```

Comment:
Can you set it to false in the others like regtest does with fPowNoRetargeting ?

Reply:
OK!